x/dex/keeper/msgserver/msg_server_register_contract.go

```diff
@@ -2,8 +2,10 @@ package msgserver
 
 import (
 	"context"
+	"errors"
```

Comment:
seems to be a lot of complex logic in here. Given that I don't expect this to run suuuuper frequently, I'd suggest adding a lot of logging so we can debug when there are issues.

Reply:
sounds good. Added a bunch of logging